### PR TITLE
HIP-584: Implement integration tests for RedirectPrecompile

### LINKED ISSUE #5651: HIP-584: Provide implementation for redirect precompile
### Problem

We need default implementations for precompiles related to redirect, so we should copy needed logic from hedera-services into hedera-mirror-web3.

### Solution

Copy precompile logic for the following precompiles:

- RedirectPrecompile

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by MrValioBg**
**Description**:

This PR adds integration tests for the RedirectPrecompile. Specifically, tests have been added for the following:

- **ERC Operations**: All Read & Modification

- **HRC Operations**:
  - HRC_ASSOCIATE
  - HRC_DISSOCIATE

**Implementation Details**:

- **Modified** the `HTSPrecompiledContract.java`: Added logic to provide the correct return signature (int32, bytes) for precompile flow during explicit Redirect calls.
- **Persisted** a new `RedirectContractTest` solidity contract in the database.
- **Implemented** ERC test cases: Found in `ContractCallServiceERCTokenTest `& HRC operations in `ContractCallServicePrecompileTest`.
- **Extracted** common test logic: Moved logic from gasEstimate tests that assert gasUsed is within the expected range into a common method in `ContractCallTestSetup`. This ensures better maintainability &  consistency.

Fixes #5651 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java [hunk 1]
```diff
@@ -283,12 +284,21 @@ public class ContractCallTestSetup extends Web3IntegrationTest {
 
     @Value("classpath:contracts/PrecompileTestContract/PrecompileTestContract.json")
     protected Path ABI_PATH;
+
+    // The contract source `RedirectTestContract.sol` is in test resources
+    @Value("classpath:contracts/RedirectTestContract/RedirectTestContract.bin")
+    protected Path REDIRECT_CONTRACT_BYTES_PATH;
```
Comment (0xivanov) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java:290]: Is this used?

Comment (MrValioBg) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java]: Removed it. Thanks :)

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java [hunk 2]
```diff
@@ -291,12 +292,17 @@ public class ContractCallTestSetup extends Web3IntegrationTest {
 
     @Value("classpath:contracts/PrecompileTestContract/PrecompileTestContract.json")
     protected Path ABI_PATH;
+
+    @Value("classpath:contracts/RedirectTestContract/RedirectTestContract.json")
+    protected Path REDIRECT_CONTRACT_ABI_PATH;
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java:297]: We should persist a contract in DB with the runtime bytecode as we do for the other contracts. Then, I suspect the tests for `redirectForToken` would fail, since we're missing an implementation for the `RedirectPrecompile`.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java (hunk 1 above)
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java]: We should use it to save the runtime bytecode of this contract into the DB.

File: hedera-mirror-test/src/test/resources/solidity/contracts/RedirectTestContract.sol
Comment (steven-sheehy) [inline hedera-mirror-test/src/test/resources/solidity/contracts/RedirectTestContract.sol:1]: Seems like this file was accidentally duplicated here and should be removed.

Comment (MrValioBg) [reply hedera-mirror-test/src/test/resources/solidity/contracts/RedirectTestContract.sol:1]: Great catch. I removed it :)

**Review by steven-sheehy — APPROVED** [review]
LGTM, just need to fix merge conflicts after removing type from `EntityId.of()`.

**Review by stoyanov-st — APPROVED** [review]
LGTM

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java [hunk 3]
```diff
@@ -707,44 +724,51 @@ protected void persistEntities() {
...
-        tokenAccountPersist(modificationContract, nftEntityId, TokenFreezeStatusEnum.UNFROZEN);
-        tokenAccountPersist(treasuryEntityId, notFrozenFungibleTokenEntityId, TokenFreezeStatusEnum.UNFROZEN);
-        tokenAccountPersist(nestedContractId, nftEntityId, TokenFreezeStatusEnum.UNFROZEN);
-        tokenAccountPersist(nestedContractId, tokenEntityId, TokenFreezeStatusEnum.UNFROZEN);
-        tokenAccountPersist(nestedContractId, nftEntityId3, TokenFreezeStatusEnum.UNFROZEN);
-        tokenAccountPersist(nestedContractId, tokenTreasuryEntityId, TokenFreezeStatusEnum.UNFROZEN);
-        tokenAccountPersist(ercContract, tokenEntityId, TokenFreezeStatusEnum.UNFROZEN);
-        tokenAccountPersist(ercContract, nftEntityId, TokenFreezeStatusEnum.UNFROZEN);
+        tokenAccountPersist(modificationContract, tokenEntityId, TokenFreezeStatusEnum.UNFROZEN, true);
+        tokenAccountPersist(modificationContract, tokenEntityIdNotAssociated, TokenFreezeStatusEnum.UNFROZEN, false);
```
Comment (0xivanov) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java:737]: You can remove this line and this will make the tokenEntityIdNotAssociated not associated. This way you can revert the changes for tokenAccountPersist. This way the PR is smaller and we will have less merge conflicts

Comment (MrValioBg) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallTestSetup.java]: Yep thats good, thanks :) Its done.

**Review by 0xivanov — APPROVED** [review]
LGTM
